### ISSUE TITLE
localisation of external projects broken

### ISSUE THREAD

**Issue by CarolingerSeilchenspringer** [liqd/a4-meinberlin#2805]
**URL:** https://meinberlin-dev.liqd.net/dashboard/organisations/caros-testorga/external-projects/
**user:** admin
**expected behaviour:** i can localize my project
**behaviour:** I can't put a pin in localisation (Verortung) of my external project. 
**important screensize:**
**device & browser:** 
**Comment/Question:** Strangely this is only true for my organisation "Caros Testorga", having the same project in "liqd" it works. Both times I was logged in as admin. Since the problem first was noticed on productive (https://mein.berlin.de/dashboard/projects/unterschutzstellungsverfahren-nsg-biesenhorster-sa/point/) it shouldn't have to do with the new dashboard. 

I could set the pin via django and there you can see one difference: the broken project has something written for the coordinates, here for the linked project above:
"{\"type\":\"Feature\",\"properties\":{},\"geometry\":{\"type\":\"Point\",\"coordinates\":[13.727478,52.451269]}}"

For the project on productive it was:
"\"{\\\"type\\\":\\\"Feature\\\",\\\"properties\\\":{},\\\"geometry\\\":{\\\"type\\\":\\\"Point\\\",\\\"coordinates\\\":[13.550488,52.481805]}}\"" 

I also compared the date of creation or phase or if project is published or not but it all doesn't seem to matter. Can you find the problem?

**Comment by fuzzylogic2000** [comment]
So, this happened in one external project on prod. And in one of the external projects on caros testorga? Can you reproduce it? Does it happen, when you add a new external project or does it occur in old ones?

**Comment by fuzzylogic2000** [comment]
For the project that doesn't work on dev (https://meinberlin-dev.liqd.net/dashboard/projects/caros-externes-projekt/external/, https://meinberlin-dev.liqd.net/django-admin/a4projects/project/292/change/?_changelist_filters=organisation__id__exact%3D14), it has actually the same strange escaping as on prod: 

"{\"type\":\"Feature\",\"properties\":{},\"geometry\":{\"type\":\"Point\",\"coordinates\":[13.727478,52.451269]}}"

**Comment by CarolingerSeilchenspringer** [comment]
mine was old (20.6.2019), the one on prod was from 7.02.2020. But another one on prod from 15.1.2020  (also external) did work according to client. New external project on Caros Testorga works. Just found another broken one on prod: https://mein.berlin.de/django-admin/a4projects/project/588/change/ created 29.08.2019 with the following strange content: "\"\\\"\\\\\\\"\\\\\\\\\\\\\\\"\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\"\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\"{\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\"type\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\":\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\"Feature\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\",\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\"properties\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\":{},\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\"geometry\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\":{\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\"type\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\":\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\"Point\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\",\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\"coordinates\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\":[13.40611,52.519276]}}\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\"\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\"\\\\\\\\\\\\\\\"\\\\\\\"\\\"\""

**Comment by CarolingerSeilchenspringer** [comment]
just managed to break a working project by changing the phase AFTER setting the pin.

**Comment by CarolingerSeilchenspringer** [comment]
@rmader can you have a look at it in this sprint and see if this is an easy fix or not?

**Comment by rmader** [comment]
I'll try - it appears to be something that, if not fixed, will strike us again and again quite badly :/

**Comment by rmader** [comment]
I couldn't reproduce yet but https://code.djangoproject.com/ticket/27183#comment:6 looks very similar, which again points to https://code.djangoproject.com/ticket/27675#comment:8:

> Django's JSONField cannot be mixed with django-jsonfield's JSONField to use. 

Here it gets a bit complicated because there are several packages with that name - however, we are using `jsonfield` in version 2.0.2 on master, which was unmaintained since mid 2017 (pre django 2). Since this february a new maintainer, who previously created a fork of the package, was granted ownership of the repo so now we have a maintained and updated package.

Long story short - I *think* the problem was simply caused by an outdated / unmaintained package which now got fixed up.

**Comment by fuzzylogic2000** [comment]
@rmader Nice! So we hope for the best and close this issue?! 
@CarolingerSeilchenspringer You broke a project by changing the phase after setting the pin? Maybe that's worth trying on stage. If it still breaks the project there, and if it's fixed after the release or already fixed on dev.

**Comment by rmader** [comment]
I now went through all projects in the DB by hand and fixed the broken ones. Will have a short look after he next release if anything new appeared.

In case anything went wrong, here's a backup file with all broken coordinate entries (they are public after all): 
[backup_projekts.txt](https://github.com/liqd/a4-meinberlin/files/4475903/backup_projekts.txt)

Therefore closing for now.

**Comment by rmader** [comment]
Huh did I not send my comment or did I remove it again? Anyhow, went through the db by hand and fixed all projects. Here's a backup of the broken coordinates in case something went wrong: 
[backup_projekts.txt](https://github.com/liqd/a4-meinberlin/files/4476026/backup_projekts.txt)
